apps/mobile-wallet/src/features/buy/BuyModal.tsx

```diff
@@ -0,0 +1,150 @@
+      const onAndroidBackPress = () => {
+        if (webViewRef.current) {
+          webViewRef.current.goBack()
+          return true // prevent default behavior (exit app)
+        }
+        return false
+      }
+
+      BackHandler.addEventListener('hardwareBackPress', onAndroidBackPress)
+      return () => {
+        BackHandler.removeEventListener('hardwareBackPress', onAndroidBackPress)
+      }
+    }
+  }, [])
+
+  if (!defaultAddress) return null
+
+  const banxaURL =
+    'https://alephium.banxa-sandbox.com/' +
+    `?walletAddress=${defaultAddress}` +
+    `&theme=${theme.name}` +
+    `&backgroundColor=${theme.bg.primary.slice(1)}` +
+    `&textColor=${theme.font.primary.slice(1)}` +
+    `&primaryColor=${theme.global.accent.slice(1)}` +
+    `&secondaryColor=${theme.global.complementary.slice(1)}`
+
+  const handleModalCloseWhenFinished = (e: WebViewNavigation) => {
+    // Close modal if url equals default return URL configured in Banxa dashboard
+    if (e.url.includes('alephium.org')) {
+      navigation.navigate('TransfersScreen')
+      props.onClose()
+    }
+  }
+
+  return (
+    <Portal>
+      <BottomModal
+        {...props}
+        Content={(props) => (
+          <ModalContent {...props} contentContainerStyle={{ flex: 1, paddingTop: 0 }}>
+            {!isDisclaimerAccepted && (
+              <DisclaimerContent>
+                <ScreenTitle title={t('Disclaimer')} />
+                <AppText style={{ flex: 1 }}>
+                  <Trans
+                    t={t}
+                    i18nKey="banxaDisclaimer"
+                    components={{
+                      1: <LinkToWeb url="https://www.banxa.com" />
+                    }}
+                  >
+                    {
+                      'You are about to access 3rd party services provided by <1>Banxa.com</1> through an in-app browser. Alephium does not control Banxa’s services. Banxa’s terms and conditions will apply, so please read and understand them before proceeding.'
+                    }
+                  </Trans>
+                </AppText>
+                <Button
+                  title={t("Alright, let's get to it.")}
+                  onPress={() => setIsDisclaimerAccepted(true)}
+                  variant="highlight"
+                  style={{ marginBottom: insets.bottom }}
+                />
+              </DisclaimerContent>
+            )}
+            <WebView
+              ref={webViewRef}
+              source={{
+                uri: banxaURL
+              }}
+              allowsInlineMediaPlayback
+              enableApplePay
+              mediaPlaybackRequiresUserAction={false}
+              containerStyle={{ padding: 0 }}
+              allowsBackForwardNavigationGestures
+              onNavigationStateChange={handleModalCloseWhenFinished}
+              nestedScrollEnabled
+            />
+          </ModalContent>
+        )}
+        title={t('Buy')}
+        maximisedContent
+        noPadding
```

Comment:
super tiny nit: I'd move these props to the top because they are kinda lost down there.